guides/source/caching_with_rails.md

```diff
@@ -286,90 +349,174 @@ simply be explicit in a comment, like:
 
-Instead, cache the ID or some other primitive data type. For example:
+```yaml
+production:
+  primary: &primary_production
+    <<: *default
+    database: app_production
+    username: app
+    password: <%= ENV["APP_DATABASE_PASSWORD"] %>
+  cache:
+    <<: *primary_production
+    database: app_production_cache
+    migrations_paths: db/cache_migrate
+```
 
-```ruby
-# super_admins is an expensive SQL query, so don't run it too often
-ids = Rails.cache.fetch("super_admin_user_ids", expires_in: 12.hours) do
-  User.super_admins.pluck(:id)
-end
-User.where(id: ids).to_a
+If `database` or [`databases`](#sharding-the-cache) is not specified in the cache configuration, Solid Cache will use the ActiveRecord::Base connection pool. This means that cache reads and writes will be part of any wrapping database transaction.
+
+In production, the cache store is configured to use the solid cache store:
+
+```yaml
+  # config/environments/production.rb
+  config.cache_store = :solid_cache_store
 ```
 
-### SQL Caching
+You can access the cache by calling `Rails.cache`.
 
-Query caching is a Rails feature that caches the result set returned by each
-query. If Rails encounters the same query again for that request, it will use
-the cached result set as opposed to running the query against the database
-again.
+### Customizing the Cache Store
 
-For example:
+Solid Cache can be customized through the config/cache.yml file or .
 
-```ruby
-class ProductsController < ApplicationController
-  def index
-    # Run a find query
-    @products = Product.all
+```yaml
+default: &default
+  store_options:
+    # Cap age of oldest cache entry to fulfill retention policies
+    max_age: <%= 60.days.to_i %>
+    max_size: <%= 256.megabytes %>
+    namespace: <%= Rails.env %>
+```
 
-    # ...
+For the full list of keys for store_options see [Cache configuration](https://github.com/rails/solid_cache#cache-configuration).
 
-    # Run the same query again
-    @products = Product.all
-  end
-end
+Here, you can adjust the `max_age` and `max_size` options to control the age and size of the cache entries.
+
+### Handling Cache Expiration
+
+Solid Cache tracks cache writes by incrementing a counter with each write. When the counter reaches 50% of the `expiry_batch_size` from the [Cache configuration](https://github.com/rails/solid_cache#cache-configuration), a background task is triggered to handle cache expiry. This approach ensures cache records expire faster than they are written when the cache needs to shrink.
+
+The background task only runs when there are writes, so the process stays idle when the cache is not being updated. If you prefer to run the expiry process in a background job instead of a thread, set `expiry_method` from the[Cache configuration](https://github.com/rails/solid_cache#cache-configuration) to `:job`, which will enqueue a `SolidCache::ExpiryJob`.
```

Comment:
Should we link the `SolidCache::ExpiryJob`?

Reply:
I stand to be corrected, but I don't think there exists a link to that?